File: src/main/python/PermutationInString_567.py
class PermutationInString_567:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        s1Code = [0]*26
        for c in s1:
            s1Code[ord(c) - ord('a')] += 1
        i = 0
        j = len(s1) - 1
        s2Code = [0]*26
        for c in s2[i:j+1]:
            s2Code[ord(c) - ord('a')] += 1
        if s1Code == s2Code:
            return True
        j += 1
        while j < len(s2):
            s2Code[ord(s2[i]) - ord('a')] -= 1
            i += 1
            s2Code[ord(s2[j]) - ord('a')] += 1
            if s1Code == s2Code:
                return True
            j += 1

        return False

File: src/main/java/org/leetcode/PermutationInString_567.java
package org.leetcode;

import java.util.Arrays;

public class PermutationInString_567 {
    public boolean checkInclusion(String s1, String s2) {
        if(s1.length() > s2.length()) return false;
        int[] s1Code = new int[26];
        int[] s2Code = new int[26];

        for(int i = 0; i < s1.length(); i++){
            s1Code[s1.charAt(i) - 'a']++;
            s2Code[s2.charAt(i) - 'a']++;
        }

        if(Arrays.equals(s1Code, s2Code)) return true;
        int i = 0;
        int j = s1.length();
        while(j < s2.length()){
            s2Code[s2.charAt(i) - 'a']--;
            s2Code[s2.charAt(j) - 'a']++;
            if(Arrays.equals(s1Code, s2Code)) return true;
            i++;
            j++;
        }

        return false;
    }
}
